### PR TITLE
Add cartodb.js w/ npm auto-update via single package.json

### PR DESCRIPTION
**Pull request by sashberd**
close 6746, cc @javisantana

# Pull request template for new library

Pull request for issue: #6746
Related issue(s): # #7812 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required): https://github.com/CartoDB/carto.js
* Official website (optional, not the repository): https://carto.com/
* NPM package url (optional): https://www.npmjs.com/package/@carto/carto.js
* License and its reference: BSD-3-Clause
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 126
  * Count of stars:  399
  * Count of forks: 264
* NPM download stats (optional):
  * Downloads in the last day:
  * Downloads in the last week: 173
  * Downloads in the last month:

## Essential checklist

* [ ] I'm the author of this library
  * [ ] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [ ] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [ ] Auto-update setup
* [ ] Auto-update target/source is valid.
* [ ] Auto-update filemap is correct.

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [] Separate unrelated changes into different commits.
* [] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/cartodb.js/packages.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "cartodb.js",
+  "filename": "carto.min.js",
+  "description": "CARTO.js is a JavaScript library to create custom location intelligence applications that leverage the power of CARTO. It is the library that powers Builder and it is part of the Engine ecosystem.",
+  "keywords": ["cartodb", "Maps", "SQL"],
```
Comment (MattIPv4) [inline ajax/libs/cartodb.js/packages.json]: Could we format this nicely, with each entry indented on a new line?

File: ajax/libs/cartodb.js/packages.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "author": {
+    "name": "CartoDB",
+    "email": "wadus@cartodb.com"
+  },
+  "license": "BSD-3-Clause",
+  "npmName": "@carto/carto.js",
+  "npmFileMap": [
+    {
+      "basePath": "@carto/carto.js",
+      "files": ["carto.*.+(js|map)"]
```
Comment (MattIPv4) [inline ajax/libs/cartodb.js/packages.json]: Could we format this nicely, with each entry indented on a new line?

**Comment by javisantana** [comment]
Hey, I'm not managing this library anymore, you should try to find the right person to ping

File: ajax/libs/cartodb.js/packages.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  "homepage": "https://cartodb.com/",
+  "author": {
+    "name": "CartoDB",
+    "email": "wadus@cartodb.com"
+  },
+  "license": "BSD-3-Clause",
+  "npmName": "@carto/carto.js",
+  "npmFileMap": [
+    {
+      "basePath": "@carto/carto.js",
```
Comment (MattIPv4) [inline ajax/libs/cartodb.js/packages.json]:
```suggestion
      "basePath": "",
```

We start from within the package already?

File: ajax/libs/cartodb.js/packages.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "cartodb.js",
```
Comment (MattIPv4) [inline ajax/libs/cartodb.js/packages.json]:
```suggestion
  "name": "carto.js",
```

This library seems to go by carto.js now, not cartodb.js?

(_Directory this file is in will also need renaming_)

**Pull request merged**